tests/vfs_iterate_mount/main.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) Christian Amsüss <chrysn@fsfe.org>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * Mount and unmount a few file systems, demonstrating that
+ * vfs_iterate_mount_dirs performs as advertised.
+ *
+ * @author      Christian Amsüss <chrysn@fsfe.org>
+ */
+
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+
+#include <vfs.h>
+#include <fs/constfs.h>
+
+static constfs_file_t constfs_files[1] = {
+    /* Not completely empty -- that'd be a hassle around empty arrays and
+     * their size */
+    {
+        .path = "some-file",
+        .size = 0,
+        .data = (void*)"",
+    },
+};
+
+static constfs_t constfs_desc = {
+    .nfiles = ARRAY_SIZE(constfs_files),
+    .files = constfs_files,
+};
+
+static vfs_mount_t mount1 = {
+    .fs = &constfs_file_system,
+    .mount_point = "/const1",
+    .private_data = &constfs_desc,
+};
+
+static vfs_mount_t mount2 = {
+    .fs = &constfs_file_system,
+    .mount_point = "/const2",
+    .private_data = &constfs_desc,
+};
+
+static vfs_mount_t mount3 = {
+    .fs = &constfs_file_system,
+    .mount_point = "/const3",
+    .private_data = &constfs_desc,
+};
+
+static vfs_mount_t mount4 = {
+    .fs = &constfs_file_system,
+    .mount_point = "/const4",
+    .private_data = &constfs_desc,
+};
+
+/* Global iterator, simply because it makes the calls visually simpler;
+ * zero-initialized */
+vfs_DIR iter;
```

Comment:
```suggestion
static vfs_DIR iter;
```
but actually 

```C
static void iter_and_report(vfs_DIR *iter) {
```

would be both cleaner and easier to follow

Reply:
Good point, fixed.